feature-libs/product-configurator/rulebased/cpq/rest/converters/cpq-configurator-serializer.ts

```diff
@@ -0,0 +1,110 @@
+import { Injectable } from '@angular/core';
+import { Converter } from '@spartacus/core';
+import { Configurator } from '@spartacus/product-configurator/rulebased';
+import { Cpq } from '../cpq.models';
+
+const VALUE_SEPARATOR = ',';
+
+@Injectable()
+export class CpqConfiguratorSerializer
+  implements Converter<Configurator.Configuration, Cpq.UpdateAttribute> {
+  convert(source: Configurator.Configuration): Cpq.UpdateAttribute {
+    const attribute: Configurator.Attribute = this.findFirstChangedAttribute(
+      source
+    );
+    let updateAttribute: Cpq.UpdateAttribute;
+    if (source.updateType === Configurator.UpdateType.ATTRIBUTE_QUANTITY) {
+      updateAttribute = this.convertQuantity(attribute, source.configId);
+    } else {
+      updateAttribute = this.convertAttribute(attribute, source.configId);
+    }
+    return updateAttribute;
+  }
+
+  convertQuantity(
```

Comment:
do we want to make some of these functions as `protected`?

Reply:
picked

Reply:
Agree, good hint